app/scripts/services/rdp.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("RdpService", function(Logger) {
+    function findRDPPort (service, rdpPortNumber) {
+      var servicePorts = _.get(service, 'spec.ports');
+      return _.find(servicePorts, {targetPort: rdpPortNumber});
+    }
+
+    return {
+      buildRdp: function (address, port) {
+        return '' +
+          'full address:s:' + address + ':' + port +
+          '\nusername:s:Administrator' +
+          '\nscreen mode id:i:2' + // set 2 for full screen
+          '\nprompt for credentials:i:1' +
+          '\ndesktopwidth:i:0' +
+          '\ndesktopheight:i:0' +
+          '\nauthentication level:i:2' +
+          '\nredirectclipboard:i:1' +
+          '\nsession bpp:i:32' +
+          '\ncompression:i:1' +
+          '\nkeyboardhook:i:2' +
+          '\naudiocapturemode:i:0' +
+          '\nvideoplaybackmode:i:1' +
+          '\nconnection type:i:2' +
+          '\ndisplayconnectionbar:i:1' +
+          '\ndisable wallpaper:i:1' +
+          '\nallow font smoothing:i:1' +
+          '\nallow desktop composition:i:0' +
+          '\ndisable full window drag:i:1' +
+          '\ndisable menu anims:i:1' +
+          '\ndisable themes:i:0' +
+          '\ndisable cursor setting:i:0' +
+          '\nbitmapcachepersistenable:i:1' +
+          '\naudiomode:i:0' +
+          '\nredirectcomports:i:0' +
+          '\nredirectposdevices:i:0' +
+          '\nredirectdirectx:i:1' +
+          '\nautoreconnection enabled:i:1' +
+          '\nnegotiate security layer:i:1' +
+          '\nremoteapplicationmode:i:0' +
+          '\nalternate shell:s:' +
+          '\nshell working directory:s:' +
+          '\ngatewayhostname:s:' +
+          '\ngatewayusagemethod:i:4' +
+          '\ngatewaycredentialssource:i:4' +
+          '\ngatewayprofileusagemethod:i:0' +
+          '\npromptcredentialonce:i:1' +
+          '\nuse redirection server name:i:0' +
+          '\n';
+      },
+
+      fileDownload: function (data, fileName, mimeType) {
+        fileName = fileName || "vm.rdp";
```

Comment:
Nit: single quotes in JS, double quotes in html.

Reply:
done